src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -1317,13 +1317,62 @@ Namespace Microsoft.CodeAnalysis.Semantics
-            Dim expression As Lazy(Of IOperation) = New Lazy(Of IOperation)(Function() Create(boundRaiseEventStatement.EventInvocation))
+        Private Function CreateBoundRaiseEventStatementOperation(boundRaiseEventStatement As BoundRaiseEventStatement) As IOperation
             Dim syntax As SyntaxNode = boundRaiseEventStatement.Syntax
             Dim type As ITypeSymbol = Nothing
             Dim constantValue As [Optional](Of Object) = New [Optional](Of Object)()
             Dim isImplicit As Boolean = boundRaiseEventStatement.WasCompilerGenerated
-            Return New LazyExpressionStatement(expression, _semanticModel, syntax, type, constantValue, isImplicit)
+
+            Dim eventInvocation = TryCast(boundRaiseEventStatement.EventInvocation, BoundCall)
+
+            ' Return an invalid statement for invalid raise event statement
+            If eventInvocation Is Nothing OrElse eventInvocation.ReceiverOpt Is Nothing
+                Debug.Assert(boundRaiseEventStatement.HasErrors)
+                Dim children As Lazy(Of ImmutableArray(Of IOperation)) = New Lazy(Of ImmutableArray(Of IOperation))(Function() ImmutableArray.Create(Of IOperation)(Create(boundRaiseEventStatement.EventInvocation)))
+                return New LazyInvalidStatement(children, _semanticModel, syntax, type, constantValue, isImplicit)
+            End If 
+            
+            Dim receiver = eventInvocation.ReceiverOpt
+            Dim eventSymbol = boundRaiseEventStatement.EventSymbol
+            Dim eventReferenceSyntax = receiver.Syntax
+            Dim eventReferenceType As ITypeSymbol = eventSymbol.Type
+            Dim eventReferenceConstantValue As [Optional](Of Object) = ConvertToOptional(receiver.ConstantValueOpt)
+            ' EventReference in a raise event statement is never implicit. However, the way it is implemented, we don't get
+            ' a "BoundEventAccess" for either field backed event or custom event, and the bound nodes we get are marked as 
+            ' generated by compiler. As a result, we have to explicitly set IsImplicit to false.
+            dim eventReferenceIsImplicit as Boolean = False
+            
+            Dim boundInstance As BoundNode
+            If receiver.Kind = BoundKind.FieldAccess
+                ' For raising a field backed event, we will only get a field access node in bound tree.
+                Dim eventFieldAccess = CType(receiver, BoundFieldAccess)
+                Debug.Assert(eventFieldAccess.FieldSymbol.AssociatedSymbol = eventSymbol)
+
+                boundInstance = eventFieldAccess.ReceiverOpt
+            Else 
+                ' This is for a custom event
+                boundInstance = receiver
+            End If
+
+            Dim eventReferenceInstance As Lazy(Of IOperation) = New Lazy(Of IOperation)(Function() If(eventSymbol.IsShared, Nothing, Create(boundInstance)))
```

Comment:
>eventSymbol.IsShared [](start = 102, length = 20)

I am not sure if we should drop the receiver just because the event is shared. It is not an error to provide receivers for shared members in VB.

Reply:
It looks like this is not going to be a problem here because the receiver cannot be specified explicitly in RaiseEvent statement. However, if you are coping a pattern from other places in the code, please open an issue to follow up on those places with targeted tests and necessary fixes.

---
In reply to: [143086345](https://github.com/dotnet/roslyn/pull/22243#discussion_r143086345) [](ancestors = 143086345)